packages/cli/src/upgrade.ts

```diff
@@ -96,12 +99,45 @@ export async function upgradeDependency([name, tag = "latest"]: string[]) {
 
 const npmRequestLimit = pLimit(40);
 
-function getPackageInfo(pkgName: string) {
-  return npmRequestLimit(async () => {
-    const getPackageJson = (await import("package-json")).default;
-
-    return getPackageJson(pkgName, {
-      allVersions: true,
+function getPackageInfo(packageName: string): Promise<PackageMetadata> {
+  return npmRequestLimit(async (): Promise<PackageMetadata> => {
+    const scope = packageName.split("/")[0];
+    const registryUrl_ = registryUrl(scope);
+    const packageUrl = new URL(
+      encodeURIComponent(packageName).replace(/^%40/, "@"),
+      registryUrl_
+    );
+    const authInfo = registryAuthToken(registryUrl_.toString(), {
+      recursive: true,
     });
+
+    const headers: Record<string, string> = {
+      accept:
+        "application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*",
+    };
+    if (authInfo) {
+      headers.authorization = `${authInfo.type} ${authInfo.token}`;
+    }
+
+    const response = await fetch(packageUrl, { headers });
```

Comment:
This is totally something I'd prefer not to have in the code 😢 I'm happy to offload this kind of code to dependencies just so I can stay ignorant of the requirements for fetching this thing.

At the very least, this should be moved to a separate function and some resources should be added in the comment to explain what's happening (I suspect most of this comes from the `package-json` package, if that's the case we should have proper license attribution for that package).

Is `package-json` maintained? Could PRs be open against that project to shrink its size?

Reply:
It is maintained, but it's ESM-only and `node >=18.18.0` only.

The latest version is significantly smaller, because it uses `ky` instead of `got` for network requests

Reply:
The CLI package here is still on 0.x version and you have already scheduled some minor releases for it (which are effectively treated as breaking for 0.x packages). Let's bump the required node version for the CLI alone

Reply:
What about `package-json` being ESM-only? I don't know of a clean solution, other than converting the cli package to ESM.

Reply:
Since this is a CLI you could also convert it to ESM but this will likely be more complicated given the test setup, tsconfigs and all. So the dynamic import parth forward (for now) is probably just simpler

Reply:
Roger, will look into this!

Reply:
OK, I needed to change `babel.config.js` to make `preconstruct dev` work. Other than that, using dynamic import() was absolutely the correct call, thanks!

Should be good for re-review, I've also updated the PR description.